invenio_vocabularies/api.py

```diff
@@ -9,82 +9,82 @@
+        return self._get_l10n("title", locale)
 
-#     def dump(self, locale):
-#         """Returns a localized copy of this object."""
-#         return {
-#             **self,
-#             "description": self.get_description(locale),
-#             "title": self.get_title(locale),
-#         }
+    def dump(self, locale):
+        """Returns a localized copy of this object."""
+        return {
+            **self,
+            "description": self.get_description(locale),
+            "title": self.get_title(locale),
+        }
 
 
-# class VocabularyBackend:
-#     """A backend implementation for a vocabulary type."""
+class VocabularyBackend:
+    """A backend implementation for a vocabulary type."""
 
-#     record_cls = Vocabulary
+    record_cls = Vocabulary
 
-#     def get(self, identifier):
-#         """Returns the vocabulary item matching this identifier."""
-#         return VocabularyItem(self.record_cls.pid.resolve(identifier))
+    def get(self, identifier):
+        """Returns the vocabulary item matching this identifier."""
+        return VocabularyItem(self.record_cls.pid.resolve(identifier))
 
-#     def get_all(self, vocabulary_type_name):
-#         """Returns all the vocabulary of this type. Potentially costly."""
-#         # TODO implement
-#         pass
+    def get_all(self, vocabulary_type_name):
+        """Returns all the vocabulary of this type. Potentially costly."""
+        # TODO implement
+        pass
 
 
-# class ResourceTypeVocabulary:
-#     """An API for a vocabulary type."""
+class ResourceTypeVocabulary:
+    """An API for a vocabulary type."""
 
-#     title = _("Resource Type")
-#     backend = VocabularyBackend()
+    title = _("Resource Type")
+    backend = VocabularyBackend()
 
-#     def __init__(self, vocabulary_type):
-#         """Constructs a new vocabulary type."""
-#         super().__init__()
-#         self.vocabulary_type = vocabulary_type
+    def __init__(self, vocabulary_type):
+        """Constructs a new vocabulary type."""
+        super().__init__()
+        self.vocabulary_type = vocabulary_type
 
-#     def get(self, identifier):
-#         """Returns the vocabulary item matching this identifier."""
-#         return self.backend.get(identifier)
+    def get(self, identifier):
+        """Returns the vocabulary item matching this identifier."""
+        return self.backend.get(identifier)
 
-#     def get_all(self):
-#         """Returns all the vocabulary of this type."""
-#         return self.backend.get_all(self.vocabulary_type)
+    def get_all(self):
+        """Returns all the vocabulary of this type."""
+        return self.backend.get_all(self.vocabulary_type)
 
-#     def dump_all(self):
-#         """Dumps all the vocabulary of this type."""
-#         return list(map(lambda v: v.dump(), self.get_all()))
+    def dump_all(self):
+        """Dumps all the vocabulary of this type."""
+        return list(map(lambda v: v.dump(), self.get_all()))
```

Comment:
minor: Huh, I didn't know about `map`. Thanks for exposing that... I still prefer list comprehensions: 

```suggestion
        return [v.dump() for v in self.get_all()]
```

Reply:
This code will be changed during the next month RDM sprint. It was not clear that it was needed before, and now I am not convinced it is the correct way of doing it. I "just" uncommented existing code. Wouldn't spend time on perfecting something that might disappear in a week time 😓 WDYT? Same for tests